lib/engine/src/artifact.rs

```diff
@@ -103,6 +103,20 @@ pub trait Artifact: Send + Sync + Upcastable {
             self.table_styles(),
         )
         .map_err(InstantiationError::Link)?;
+
+        let mut thunks = vec![];
+        // ------------
+        for (func_init, func) in imports
+            .host_function_env_initializers
+            .values()
+            .cloned()
+            .zip(imports.functions.values())
+        {
+            let host_env = func.vmctx as _;
```

Comment:
Hmm I predict complex bug with this line in the future 🤔.

Reply:
Yeah in #1753 I'm trying to make use of `vmctx` more explicit so that users of the code are more careful with it...

Any suggestions for this use ?